src/lib.rs

```diff
@@ -885,51 +925,747 @@ impl GooseAttack {
+                    detail: "The --throttle-requests option can not be set together with the --manager flag.".to_string(),
+                });
+            }
+            // Be sure throttle_requests is in allowed range.
+            if throttle_requests == 0 {
+                return Err(GooseError::InvalidOption {
+                    option: "--throttle-requests".to_string(),
+                    value: throttle_requests.to_string(),
+                    detail: "The --throttle-requests option must be set to at least 1 request per second.".to_string(),
+                });
+            } else if throttle_requests > 1_000_000 {
+                return Err(GooseError::InvalidOption {
+                    option: "--throttle-requests".to_string(),
+                    value: throttle_requests.to_string(),
+                    detail: "The --throttle-requests option can not be set to more than 1,000,000 requests per second.".to_string(),
+                });
+            }
+            throttle_requests
+        } else if let Some(throttle_requests) = self.defaults.throttle_requests {
+            if self.attack_mode == GooseMode::Manager {
+                0
+            } else {
+                // Be sure throttle_requests is in allowed range.
+                if throttle_requests == 0 {
+                    return Err(GooseError::InvalidOption {
+                        option: "GooseDefault::ThrottleRequests".to_string(),
+                        value: throttle_requests.to_string(),
+                        detail: "The GooseDefault::ThrottleRequests default must be set to at least 1 request per second.".to_string(),
+                    });
+                } else if throttle_requests > 1_000_000 {
+                    return Err(GooseError::InvalidOption {
+                        option: "GooseDefault::ThrottleRequests".to_string(),
+                        value: throttle_requests.to_string(),
+                        detail: "The GooseDefault::ThrottleRequests default can not be set to more than 1,000,000 requests per second.".to_string(),
+                    });
+                }
+                throttle_requests
+            }
+        } else {
+            0
+        };
+
+        if self.throttle_requests > 0 {
+            info!("throttle_requests = {}", self.throttle_requests);
+        }
+
+        Ok(())
+    }
+
+    // Determine if no_reset_statics is enabled.
+    fn no_reset_metrics(&self) -> bool {
+        if self.configuration.no_reset_metrics {
+            true
+        } else if let Some(default) = self.defaults.no_reset_metrics {
+            // Do not default to no_reset_metrics on Worker.
+            if self.attack_mode == GooseMode::Worker {
+                false
+            } else {
+                default
+            }
+        } else {
+            false
+        }
+    }
+
+    // Determine if the status_codes flag is enabled.
+    fn set_status_codes(&mut self) {
```

Comment:
@LionsAd In this case, we're overloading the `configuration` (as we need this information on Workers)